src/execution_plans/metrics.rs

```diff
@@ -0,0 +1,512 @@
+            query_id,
+            num: 1,
+            name: "test_stage".to_string(),
+            plan: limit_exec.clone(),
+            inputs: vec![Arc::new(child_stage)],
+            tasks: vec![ExecutionTask {
+                url: None,
+                partition_group: vec![0],
+            }],
+            depth: 0,
+        }
+    }
+
+    /// creates a "distinct" set of metrics from the provided seed
+    fn make_distinct_metrics_set(seed: u64) -> MetricsSetProto {
+        const TEST_TIMESTAMP: i64 = 1758200400000000000; // 2025-09-18 13:00:00 UTC
+        MetricsSetProto {
+            metrics: vec![
+                MetricProto {
+                    metric: Some(MetricValueProto::OutputRows(OutputRows { value: seed })),
+                    labels: vec![],
+                    partition: None,
+                },
+                MetricProto {
+                    metric: Some(MetricValueProto::ElapsedCompute(ElapsedCompute {
+                        value: seed,
+                    })),
+                    labels: vec![],
+                    partition: None,
+                },
+                MetricProto {
+                    metric: Some(MetricValueProto::StartTimestamp(StartTimestamp {
+                        value: Some(TEST_TIMESTAMP + (seed as i64 * 1_000_000_000)),
+                    })),
+                    labels: vec![],
+                    partition: None,
+                },
+                MetricProto {
+                    metric: Some(MetricValueProto::EndTimestamp(EndTimestamp {
+                        value: Some(TEST_TIMESTAMP + ((seed as i64 + 1) * 1_000_000_000)),
+                    })),
+                    labels: vec![],
+                    partition: None,
+                },
+            ],
+        }
+    }
+
+    #[tokio::test]
+    async fn test_metrics_rewriter() {
+        let test_stage = make_test_stage_exec_with_3_nodes();
+        let test_metrics_sets = (0..3)
+            .map(|i| make_distinct_metrics_set(i + 10))
+            .collect::<Vec<MetricsSetProto>>();
+
+        let rewriter = TaskMetricsRewriter::new(test_metrics_sets.clone());
+        let plan_with_metrics = rewriter
+            .enrich_task_with_metrics(test_stage.plan.clone())
+            .unwrap();
+
+        let plan_str =
+            DisplayableExecutionPlan::with_full_metrics(plan_with_metrics.as_ref()).indent(true);
+        let expected = r"LocalLimitExec: fetch=10, metrics=[output_rows=10, elapsed_compute=10ns, start_timestamp=2025-09-18 13:00:10 UTC, end_timestamp=2025-09-18 13:00:11 UTC]
+  CoalesceBatchesExec: target_batch_size=1024, metrics=[output_rows=11, elapsed_compute=11ns, start_timestamp=2025-09-18 13:00:11 UTC, end_timestamp=2025-09-18 13:00:12 UTC]
+    ProjectionExec: expr=[id@0 as id, name@1 as name], metrics=[output_rows=12, elapsed_compute=12ns, start_timestamp=2025-09-18 13:00:12 UTC, end_timestamp=2025-09-18 13:00:13 UTC]
+      ArrowFlightReadExec, metrics=[]
+";
```

Comment:
Very nice metric tests with the `seed` to provide distinct metric value.

- Can we have the `r"LocalLimitExec: ....` in a new line so we can read it easier?
- Was the EmptyExec removed? For unit test, it is good enough. Do you plan to add more integration tests in following PRs? I think we can use `sqllogictest` or the like for those so we can test many queries without a lot of mock functions if that is easier for you. Actually, you can use this ticket for adding more those tests later. We want mutiple stages, multiple tasks and simple plus complicated queries to make sure all metrics are capture